Mathlib/Data/List/FinRange.lean

```diff
@@ -67,6 +72,33 @@ theorem nodup_ofFn {α n} {f : Fin n → α} : Nodup (ofFn f) ↔ Function.Injec
     exact h.imp_right ih
 #align list.nodup_of_fn List.nodup_ofFn
 
+/-- Folding over a list obtained from a function `g : Fin n → α` is equivalent to
+    folding over the list of indices `0, 1, ..., n-1` and invoking `g` for each index -/
+theorem List.foldl_ofFn {α β n} (f : α → β → α) (init : α) (g : Fin n → β) :
+    (List.ofFn g).foldl f init = (List.finRange n).foldl (fun a i => f a (g i)) init := by
+  induction' n with n ih generalizing init
+  · rfl
+  · simp [List.finRange_succ_eq_map, List.foldl_map, ih]
```

Comment:
Doesn't this follow immediately by unfolding `ofFn` and finding some lemma about `map` and `foldl`?

Reply:
Not really, `List.ofFn` is defined in terms of `Array.ofFn`, which in turn is defined as 
```lean
def ofFn {n} (f : Fin n → α) : Array α := go 0 (mkEmpty n) where
  /-- Auxiliary for `ofFn`. `ofFn.go f i acc = acc ++ #[f i, ..., f(n - 1)]` -/
  go (i : Nat) (acc : Array α) : Array α :=
    if h : i < n then go (i+1) (acc.push (f ⟨i, h⟩)) else acc
```

Reply:
Oh, I think that changed! I think there should be a lemma that says `ofFn f = (finRange n).map f` that you can use?

Reply:
Aha, indeed there is! That simplifies these proofs a lot, thanks for pointing them out